src/renderers/dom/stack/client/ReactDOMComponent.js

```diff
@@ -583,6 +584,20 @@ ReactDOMComponent.Mixin = {
         );
         didWarnShadyDOM = true;
       }
+
+      if (__DEV__) {
+        if (this._namespaceURI === DOMNamespaces.html) {
```

Comment:
I'm not sure we should add the warning if it's not consistent. Users could still mess up inside an `svg` tag and they wouldn't get any warnings, which is extra confusing since they would get warnings anywhere else. @spicyj what do you think?